src/script/sign.cpp

```diff
@@ -172,7 +173,7 @@ bool ProduceSignature(const BaseSignatureCreator& creator, const CScript& fromPu
     sigdata.scriptSig = PushAll(result);
 
     // Test solution
-    return solved && VerifyScript(sigdata.scriptSig, fromPubKey, STANDARD_SCRIPT_VERIFY_FLAGS, creator.Checker(), sigversion, serror);
+    return solved && VerifyScript(sigdata.scriptSig, fromPubKey, STANDARD_SCRIPT_VERIFY_FLAGS | SCRIPT_VERIFY_EXCHANGEADDR, creator.Checker(), sigversion, serror);
```

Comment:
You can add `SCRIPT_VERIFY_EXCHANGEADDR` to the list of `STANDARD_SCRIPT_VERIFY_FLAGS` instead of adding it externally like here